src/main/java/de/tum/in/www1/artemis/repository/specs/PostSpecs.java

```diff
@@ -0,0 +1,156 @@
+package de.tum.in.www1.artemis.repository.specs;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.criteria.*;
+
+import org.springframework.data.jpa.domain.Specification;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.DisplayPriority;
+import de.tum.in.www1.artemis.domain.enumeration.SortingOrder;
+import de.tum.in.www1.artemis.domain.metis.*;
+
+public class PostSpecs {
+
+    public static Specification<Post> getCourseSpecification(Long courseId, Long lectureId, Long exerciseId) {
+        return (root, query, criteriaBuilder) -> {
+            if (lectureId != null || exerciseId != null) {
+                return criteriaBuilder.conjunction();
```

Comment:
You can just return null :+1:

Reply:
I'm not sure why but in some cases it throws an error that way.

Reply:
mhm good to know :thinking: